Updated site-to-site IPsec VPN documentation (#1660)
vyos/vyos-documentation: docs/configuration/vpn/ipsec/ipsec_general.rst

.. _ipsec_general:

#########################
IPsec General Information
#########################

***********************
Information about IPsec
***********************

IPsec is the framework used to secure data.
IPsec accomplishes these goals by providing authentication,
encryption of IP network packets, key exchange, and key management.

> VyOS uses Strongswan package to implement IPsec.

Reviewer: ```suggestion
VyOS uses strongSwan for its IPsec implementation.
```

IKEv2 is that IKEv2 is much simpler and more reliable than IKEv1 because
fewer messages are exchanged during the establishment of the VPN and
additional security capabilities are available.

IKE Authentication
==================

VyOS supports 3 authentication methods.
    * **Pre-shared keys**: In this method, both peers of the IPsec
      tunnel must have the same preshared keys.
    * **Digital certificates**: PKI is used in this method.
    * **RSA-keys**: If the RSA-keys method is used in your IKE policy,
      you need to make sure each peer has the other peer’s public keys.

*************************
DPD (Dead Peer Detection)
*************************

This is a mechanism used to detect when a VPN peer is no longer active.
This mechanism has different algorithms in IKEv1 and IKEv2 in VyOS.
DPD Requests are sent as ISAKMP R-U-THERE messages and DPD Responses
are sent as ISAKMP R-U-THERE-ACK messages. In IKEv1, DPD sends messages
every configured interval. The remote peer is considered unreachable
if no response to these packets is received within the DPD timeout.
In IKEv2, DPD sends messages every configured interval. If one request

> is not responded, Strongswan execute its retransmission algorithm with

Reviewer: ```suggestion
does not receive a response, strongSwan executes its retransmission algorithm with
```

 * **start** - Tries to immediately re-create the CHILD_SA.

.. cfgcmd:: set vpn ipsec ike-group <name> ikev2-reauth

  Whether rekeying of an IKE_SA should also reauthenticate
  the peer. In IKEv1, reauthentication is always done.
  Setting this parameter enables remote host re-authentication
  during an IKE rekey.

.. cfgcmd:: set vpn ipsec ike-group <name> key-exchange

  Which protocol should be used to initialize the connection
  If not set both protocols are handled and connections will
  use IKEv2 when initiating, but accept any protocol version
  when responding:

 * **ikev1** - Use IKEv1 for Key Exchange.
 * **ikev2** - Use IKEv2 for Key Exchange.

.. cfgcmd:: set vpn ipsec ike-group <name> lifetime

  IKE lifetime in seconds <0-86400> (default 28800).

.. cfgcmd:: set vpn ipsec ike-group <name> mode

  IKEv1 Phase 1 Mode Selection:

 * **main** - Use Main mode for Key Exchanges in the IKEv1 Protocol
   (Recommended Default).
 * **aggressive** - Use Aggressive mode for Key Exchanges in the IKEv1
   protocol aggressive mode is much more insecure compared to Main mode.

.. cfgcmd:: set vpn ipsec ike-group <name> proposal <number> dh-group <dh-group number>

>   Dh-group. Default value is **2**.

Reviewer: ```suggestion
  Diffie-Hellman algorithm group. Default value is **2**.
```
